p2p/providers/peerstore_provider/rpc/provider.go

```diff
@@ -11,42 +11,47 @@ import (
 	"github.com/pokt-network/pocket/p2p/providers/peerstore_provider"
 	typesP2P "github.com/pokt-network/pocket/p2p/types"
 	"github.com/pokt-network/pocket/rpc"
-	"github.com/pokt-network/pocket/runtime/configs"
 	"github.com/pokt-network/pocket/shared/core/types"
 	"github.com/pokt-network/pocket/shared/modules"
 	"github.com/pokt-network/pocket/shared/modules/base_modules"
 )
 
-var _ peerstore_provider.PeerstoreProvider = &rpcPeerstoreProvider{}
+var (
+	_ modules.Submodule                    = &rpcPeerstoreProvider{}
+	_ peerstore_provider.PeerstoreProvider = &rpcPeerstoreProvider{}
+	_ rpcPeerstoreProviderFactory          = &rpcPeerstoreProvider{}
+)
+
+type rpcPeerstoreProviderOption func(*rpcPeerstoreProvider)
+type rpcPeerstoreProviderFactory = modules.FactoryWithOptions[peerstore_provider.PeerstoreProvider, rpcPeerstoreProviderOption]
 
-// TECHDEBT(#810): refactor to implement `Submodule` interface.
 type rpcPeerstoreProvider struct {
-	// TECHDEBT(#810): simplify once submodules are more convenient to retrieve.
-	base_modules.IntegratableModule
-	base_modules.InterruptableModule
+	base_modules.IntegrableModule
 
 	rpcURL    string
-	p2pCfg    *configs.P2PConfig
 	rpcClient *rpc.ClientWithResponses
 }
 
-func Create(options ...modules.ModuleOption) *rpcPeerstoreProvider {
+func Create(bus modules.Bus, options ...rpcPeerstoreProviderOption) (peerstore_provider.PeerstoreProvider, error) {
+	return new(rpcPeerstoreProvider).Create(bus, options...)
+}
+
+func (*rpcPeerstoreProvider) Create(
+	bus modules.Bus,
+	options ...rpcPeerstoreProviderOption,
+) (peerstore_provider.PeerstoreProvider, error) {
 	rabp := &rpcPeerstoreProvider{
 		rpcURL: flags.RemoteCLIURL,
 	}
+	rabp.SetBus(bus)
```

Comment:
@Olshansk this is why I didn't have the module registration function called in the TreeStore module refactor PR - I was treating the TreeStore as a submodule of Persistence, like how rpcPeerstoreProvider is a submodule of P2P, because I didn't see a reason for TreeStore to be called outside of the persistence context.

Reply:
@Olshansk - With @h5law possibly needing to access TreeStore from the IBC module, it's worth reevaluating that status and treating the TreeStore as a full module.

Reply:
`bus.RegisterModulr()` now takes a `Submodule` such that it could be used here instead. It will also set the bus on the module IIRC.

Reply:
Awesome 💯

Reply:
Appreciate the CC on all these. Makes sense!